modules/engines/engine-odata/src/main/java/org/eclipse/dirigible/engine/odata2/handler/ScriptingOData2EventHandler.java

```diff
@@ -46,46 +46,52 @@ public class ScriptingOData2EventHandler implements OData2EventHandler {
 	private static final Logger logger = LoggerFactory.getLogger(DefaultSQLProcessor.class);
 	
 	private static final String DIRIGIBLE_ODATA_WRAPPER_MODULE_ON_EVENT = "odata/wrappers/onEvent";
+
+	private static final String ERROR_EXECUTING_XSJSLIB_HANDLER = "Error executing .xsjslib handler: ";
 	
 	private ODataCoreService odataCoreService = new ODataCoreService();
 
 	@Override
 	public void beforeCreateEntity(PostUriInfo uriInfo, String requestContentType,
-			String contentType, ODataEntry entry) {
+			String contentType, ODataEntry entry, Map<Object, Object> context) throws ODataException {
 		try {
 			String namespace = uriInfo.getTargetType().getNamespace();
 			String name = uriInfo.getTargetType().getName();
 			String method = ODataHandlerMethods.create.name();
 			String type = ODataHandlerTypes.before.name();
 			List<ODataHandlerDefinition> handlers = odataCoreService.getHandlers(namespace, name, method, type);
-			Map<Object, Object> context = new HashMap<Object, Object>();
 			context.put("uriInfo", uriInfo);
 			context.put("requestContentType", requestContentType);
 			context.put("contentType", contentType);
 			context.put("entry", entry);
 			executeHandlers(handlers, context);
 		} catch (EdmException | ODataException e) {
 			logger.error(e.getMessage(), e);
+		} catch (ScriptingException e) {
+			logger.error(e.getMessage(), e);
```

Comment:
Don't log the exception and after that re-throw it. This is considered bad practice since it will appear two times in the log. Someone will catch and log the exception that you are re-throwing